src/modinfo.c

```diff
@@ -3860,38 +3860,59 @@ sr_modinfo_generate_config_change_notif(struct sr_mod_info_s *mod_info, sr_sessi
     return err_info;
 }
 
+/**
+ * @brief Learn if a new oper push entry must be created for this session and module in Ext SHM.
+ *        If entry already exists, learn if has_data flag needs to be changed.
+ *
+ * @param[in] sess Session to use.
+ * @param[in] mod_name Module name for which operational data is being pushed.
+ * @param[in] has_data Whether this session has data for this module.
+ * @param[out] create Learn whether oper push data for module must be created in Ext SHM.
+ * @param[out] change_has_data Learn whether has_data for module changed from the previous value.
+ */
+static void
+sr_modinfo_push_oper_mod_learn_changes(sr_session_ctx_t *sess, const char *mod_name, int has_data, int *create, int *change_has_data)
+{
+    uint32_t i;
+
+    for (i = 0; i < sess->oper_push_mod_count; ++i) {
+        if (!strcmp(sess->oper_push_mods[i].name, mod_name)) {
+            /* don't need to create in Ext SHM */
+            *create = 0;
+            /* learn whether has_data flag is being changed */
+            *change_has_data = (sess->oper_push_mods[i].has_data != has_data);
+            return;
+        }
+    }
+
+    /* create in Ext SHM */
+    *create = 1;
+    /* not a change */
+    *change_has_data = 0;
+}
+
 /**
  * @brief Update push oper mod data in a session, add a new module if not yet present.
  *
  * @param[in] sess Session to update.
  * @param[in] mod_name Module name.
  * @param[in] has_data Flag to store.
- * @param[out] created Learn whether oper push data for module was just created (first time) .
- * @param[out] changed_has_data Learn whether has_data for module changed from the previous value.
- * @return err_info, NULL on success.
+ * @return err_info, NULL on success, only fails if out of memory.
  */
-static sr_error_info_t *
-sr_modinfo_push_oper_mod_update_sess(sr_session_ctx_t *sess, const char *mod_name, int has_data, int *created, int *changed_has_data)
+sr_error_info_t *
```

Comment:
It seems this function can remain `static`.

Reply:
Ah, yes. let me fix that